pkg/sql/opt/norm/custom_funcs.go

```diff
@@ -504,6 +504,36 @@ func (c *CustomFuncs) ProjectColsFromBoth(left, right memo.GroupID) memo.GroupID
 	return pb.buildProjections()
 }
 
+// ProjectColMapSide implements the side-agnostic logic from ProjectColMapLeft
+// and ProjectColMapRight.
+func (c *CustomFuncs) ProjectColMapSide(toList opt.ColList, fromList opt.ColList) memo.GroupID {
```

Comment:
No need to export this method, since it's a helper. I'd make it `projectColMapSide` and move it below the other 2 functions.